Makefile

```diff
@@ -30,7 +30,7 @@ ifeq (,$(filter-out undefined default,$(origin ARFLAGS)))
 endif
 NVCC ?= $(CUDA_DIR)/bin/nvcc
 NVCC_CXX ?= $(CXX)
-HIPCC ?= $(HIP_DIR)/bin/hipcc
+HIPCC ?= $(ROCM_DIR)/bin/hipcc
```

Comment:
This will need to be updated in the README as well

Reply:
and here: https://github.com/CEED/libCEED/blob/main/.gitlab-ci.yml

Reply:
I've just a quick find/replace in 493cdfe

Reply:
Any thoughts on how to handle non-ROCm HIP? There is `/opt/rocm/hip/bin/hipcc` in addition to `/opt/rocm/bin/hipcc` on Noether. I'm not sure if `HIP_DIR = $(ROCM_DIR)` or `HIP_DIR = $(ROCM_DIR)/hip` should be used, and then we should use `HIP_DIR` for things that are not specific to ROCm?

Reply:
I am not familiar with Noether, but I found this on Crusher:
For rocm-5.1.0: `hipcc` is under `/opt/rocm-5.1.0/hip/bin`. There is a symlink under `/opt/rocm-5.1.0/bin`, which points to `../hip/bin/hipcc`.

For rocm-5.2.0, 5.3.0, and 5.4.0, it is the opposite scenario, `hipcc` is under `/opt/rocm-5.x.0/bin` and the symlinks are under `/opt/rocm-5.x.0/hip/bin`. I think staying with `ROCM_DIR` is less ambiguous than `HIP_DIR`.

I am not sure what you mean by non-ROCM HIP. Do you mean using HIP to run on NVIDIA for example?

Reply:
Yeah, HIP to run on Intel GPUs.

Reply:
IMHO, I think using `ROCM_DIR` is fairly more accurate than `HIP_DIR` when targeting AMD GPUs. If there is a plan to run on Intel GPUs using HIP, I think we need a distinctive name like `HIPSYCL_DIR` or something similar. I'm not sure if this has to be resolved in this PR though.

Reply:
Maybe we can use `ROCM_DIR` to select `HIPCC` and `HIP_DIR`/`HIP_LIB`? I'm not sure if there will be (or should be) a way to build the HIP backend for ROCm and build it for SPV, in which case there would be `/gpu/rocm` and `/gpu/spv` (or `/gpu/intel`, whatever they want to call it). But that's all getting messy/out of scope, so let's just try to make `ROCM_DIR` provide defaults and use the `HIP_*` variables in a vendor-agnostic way.